### ISSUE TITLE
[GridBundle] Impossible to create new Grid Driver

### ISSUE THREAD

**Issue by Roshyo** [Sylius/Sylius#10137]
**Sylius version affected**: 1.x (I believe all of them, from the 1.0. 1.3 for sure)

**Description**  
In order to fetch entities from an API and then  display them in a grid, we have to create at least a new Driver and register it.
The thing is, when trying to register the new Driver in SyliusGrid, this exception is thrown :

The value "your\/driver" is not allowed for path "sylius_resource.drivers.1". Permissible values: "doctrine\/orm", "doctrine\/mongodb-odm", "doctrine\/phpcr-odm"

This is due to this in Sylius\Bundle\ResourceBundle\DependencyInjection\Configuration.php
```php
...
    /**
     * @param ArrayNodeDefinition $node
     */
    private function addDriversSection(ArrayNodeDefinition $node): void
    {
        $node
            ->children()
                ->arrayNode('drivers')
                    ->defaultValue([SyliusResourceBundle::DRIVER_DOCTRINE_ORM])
                    ->enumPrototype()->values(SyliusResourceBundle::getAvailableDrivers())->end()
                ->end()
            ->end()
        ;
    }
...
```

So the first step is (do not reproduce at home) to override the availableDrivers and add our (all done in vendor...)

Then, in Sylius\Bundle\ResourceBundle\AbstractResourceBundle, there is a switch/case to also mpodify and add our new value.
Next step (nearly last one) we also have to override the switch/case in Sylius\Bundle\ResourceBundle\DependencyInjection\Driver\DriverProvider

And to finish, we are warned that the XML file of our provider does not exist. So we create a new one in 
Sylius/resource-bundle/Resources/config/services/integration/your/driver.xml

**Steps to reproduce**  
Install Sylius (GridBundle at least) and try to add a new Provider

**Possible Solution**  
Be more permissive in this bundle configuration. It is just one step away from being able to fetch data in API and display them in a grid.

**Comment by Prometee** [comment]
If someone get here is **possible** to do it but not easily.

First into your class `*Plugin extends Bundle implements ResourceBundleInterface`
add a new SupportedDrivers.

An override have to be made into the `DependencyInjection\*Extension extends extends AbstractResourceExtension` to give your own logic of how to get the `Entity`, the `Repository` and the `Manager`, then make a new `DependencyInjection\Driver` to handle each of those points.
Build the doctrine like `Driver` with 3 classes :

1.  `Driver\Driver` handling a `Registry/ManagerRegistry`,  and the `Driver/DataSource` 
2. `Driver\DataSource` (Transport of data between the API service and the view)
3. `Driver\ExpressionBuilder` manage `or` `and`, etc.. to allow you to sort and filter the results into the `Driver\DataSource`

And finally the Custom `Repository`(`ies`) extending `Sylius\Component\Resource\Repository\InMemoryRepository` and using your API service to `find`, `findBy`, etc every `Entity` you need

After that you will be able to create your `APIEntity` (like a Doctrine one), your `Form\Type` and configure your grid (using `driver: 'my_vendor/my_custom_driver'`)

I would like to be able to create a `sylius-plugin` or a `library` with an exemple to handle this specific thing, but it's not possible right now... Ping me if you want code exemples.

**Comment by pamil** [comment]
Would the similar mechanism to the one from ThemeBundle work for you: https://docs.sylius.com/en/1.3/components_and_bundles/bundles/SyliusThemeBundle/configuration_sources/creating_custom.html?

**Comment by Roshyo** [comment]
According to the doc, it requires a Bundle. Couldn't it be done without ? Symfony 4 advice to do everything in src without any bundle.
Except this point, I guess it can work.

**Comment by doctorx32** [comment]
@Roshyo You can register driver without a bundle with:
```
App\Mssql\Driver:
        tags:
            - { name: "sylius.grid_driver", alias: "mssql" }
```

**Comment by Roshyo** [comment]
I don't think so since the list of allowed drivers is fixed in the vendor files. As described above

**Comment by vvasiloi** [comment]
This should be reopened and moved to grid repository.
